sdk/iot/Azure.Iot.Hub.Service/tests/prerequisites/setup.ps1

```diff
@@ -0,0 +1,120 @@
+param(
+    [Parameter(Mandatory)]
+    [string] $Region,
+
+    [Parameter(Mandatory)]
+    [string] $ResourceGroup,
+    
+    [Parameter(Mandatory)]
+    [string] $SubscriptionId,
+
+    [Parameter()]
+    [string] $IotHubName,
+
+    [Parameter()]
+    [string] $AppRegistrationName
+)
+
+Function Connect-AzureSubscription()
+{
+    # Ensure the user is logged in
+    try
+    {
+        $azureContext = az account show
+    }
+    catch { }
+
+    if (-not $azureContext)
+    {
+        Write-Host "`nPlease login to Azure..."
+        az login
+        $azureContext = az account show
+    }
+
+    # Ensure the desired subscription is selected
+    $sub = az account show --output tsv --query id
+    if ($sub -ne $SubscriptionId)
+    {
+        Write-Host "`nSelecting subscription $SubscriptionId"
+        az account set --subscription $SubscriptionId
+    }
+
+    return $azureContext
+}
+
+$isAdmin = ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator")
+if (-not $isAdmin)
+{
+    throw "This script must be run in administrative mode."
+}
+
+Connect-AzureSubscription
+
+$Region = $Region.Replace(' ', '')
+
+if (-not $IotHubName)
+{
+    $IotHubName = $ResourceGroup
+}
+
+if (-not $AppRegistrationName)
+{
+    $AppRegistrationName = $ResourceGroup
+}
+
+$appId = az ad app list --show-mine --query "[?displayName=='$AppRegistrationName'].appId" --output tsv
```

Comment:
Question - is this for the storage service's identity based jobs API tests, or does IoT hub already have identity based authentication configured?

Reply:
I have not yet done any validation as to what we need the identity based auth for, this just makes sure you can setup an app registration and service principal as part of your setup or use the one you already have setup.